.github/workflows/sprd.yaml

```diff
@@ -0,0 +1,37 @@
+name: Spread 
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+  GCE:
+    runs-on: [self-hosted]
```

Comment:
out of curiosity what connects the github action runs from this repo to the self-hosted thing. Is there config on the repo itself, or do you just provide some token/auth data to the self-hosted worker when you start it up? If this setup is security sensitive, we can also chat offline tomorrow too

Reply:
This is a property of the repository. We have registered self-hosted workers that talk to GitHub and wait for requests. Each worker runs in a rootless container, without ability to run sudo (inside the container). The authentication is indeed based on a token that was obtained when each worker was registered.

Reply:
This process is explained in detail in https://help.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners